homeassistant/components/airzone_cloud/climate.py

```diff
@@ -290,9 +314,58 @@ def __init__(
         ]
         if HVACMode.OFF not in self._attr_hvac_modes:
             self._attr_hvac_modes += [HVACMode.OFF]
+        if (
+            self.get_airzone_value(AZD_SPEED) is not None
+            and self.get_airzone_value(AZD_SPEEDS) is not None
+        ):
+            self._initialize_fan_speeds()
 
         self._async_update_attrs()
 
+    def _initialize_fan_speeds(self) -> None:
+        """Initialize Aidoo fan speeds."""
+        azd_speeds: dict[int, int] = self.get_airzone_value(AZD_SPEEDS)
+        max_speed = max(azd_speeds)
+
+        fan_speeds: dict[int, str]
+        if speeds_map := FAN_SPEED_MAPS.get(max_speed):
+            fan_speeds = speeds_map
+        else:
+            fan_speeds = {}
+
+            for speed in azd_speeds:
+                if speed == 0:
+                    fan_speeds[speed] = FAN_AUTO
+                else:
+                    fan_speeds[speed] = f"{int(round((speed * 100) / max_speed, 0))}%"
+
+            fan_speeds[1] = FAN_LOW
+            fan_speeds[int(round((max_speed + 1) / 2, 0))] = FAN_MEDIUM
+            fan_speeds[max_speed] = FAN_HIGH
+
+        if 0 in fan_speeds and 0 not in azd_speeds:
+            fan_speeds.pop(0)
```

Comment:
When does this happen, is it if we pick an item in `FAN_SPEED_MAPS` and the device does not have an automatic speed? Please add a comment.

Reply:
Exactly, it's only for the case that you've mentioned.
I can either add the comment or refactor the code removing 0 from FAN_SPEED_MAPS and adding it only if the device supports the auto speed.

Reply:
Done in 2d63accac765cc94515ca58f2e8ce0af3eef493d
Please let me know if it's more clear now or if we still need a comment for this.

Reply:
`max_speeds` can be `2` or `3` I guess. But `int(round((max_speed + 1) / 2, 0))` is always `2` if `max_speed` is `2` or `3`. May be the code can be made much easier to read?